Data files from docs.cask.co

awholegunch opened the pull request:
#New file is docs.cask.co/www/resources/scripts/redirect-page.js
Modified file is docs.cask.co/www/404.html which has been changed to use it

The redirect page and the modified 404 are to address https://issues.cask.co/browse/INFRA-31
The Error Document for the bucket is (and should be) set to this modified 404.html page.

Results at http://builds.cask.co/browse/DOC-SITE0-19/artifact

Commit:
Data files from docs.cask.co
New file is docs.cask.co/www/resources/scripts/redirect-page.js
Modified file is docs.cask.co/www/404.html which has been changed to use it

Commit:
Handle directories above the numbered versions.

Commit:
Fix titles.

wolf31o2 commented on docs.cask.co/www/404.html:
```diff
@@ -0,0 +1,169 @@
+    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+    '//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+    })(window,document,'script','dataLayer','GTM-KWLFGH');</script>
+    <!-- Google Tag Manager end -->
```
Isn't this block stored in another file we can include, instead?

awholegunch replied:
Thought this block is included in just about all of our files, I don't know if it can be separate. Looking at Google's docs, they _never_ suggest including it that way. It might have something to do with latency and the asynchronous nature of that code block; it might need to be directly in the page rather than in a separate file. This was included following their directions.

wolf31o2 replied:
OK. For some reason I was thinking that it was included in our docs, not in-line, but I think I am just mistaken.

awholegunch replied:
No, you're not mistaken: it is included in our docs. But there it is the template file that Sphinx then uses to build the documentation. So each HTML page it as it is here. It is included in `~/Source/cdap/cdap-docs/_common/_themes/cdap/layout.html`.

wolf31o2 commented on docs.cask.co/www/404.html:
```diff
@@ -0,0 +1,169 @@
+    </script>
+    <script type="text/javascript" src="cdap/current/en/_static/jquery.js"></script>
+    <script type="text/javascript" src="cdap/current/en/_static/underscore.js"></script>
+    <script type="text/javascript" src="cdap/current/en/_static/doctools.js"></script>
```
Should any of this be at the root level? Maybe under `resources/scripts`? Seems like yes, to me.

awholegunch replied:
I've made copies of these in the `resources` directory, and linked to them in the upper-level files as appropriate. There are still copies in the `current` directories, as per the comment below.

wolf31o2 commented on docs.cask.co/www/resources/scripts/redirect-page.js:
```diff
@@ -0,0 +1,44 @@
+ */
+
+(function() {
+  var products = new Array( "cdap", "coopr", "tigon" );
```
Can we read this list from a file that we generate? This could be generated by the `TYPES` list that gets run in the build.

awholegunch replied:
The browser wouldn't be able to read the file, but we could generate this file (redirect-page.js) dynamically. I've added that to the `build.sh` script.

wolf31o2 reviewed (changes requested):
A few minor changes, and a request to reduce duplication. Do we need multiple copies of jquery.js or the `cask_404_75pc.png` file? Can they not just go under `/resources` and all the docs that are on the site use them?

Commit:
Change index files to disable them.

Commit:
Update case of references.

Commit:
Change to having an internal www directory inside the target directory.

Commit:
Throw in a zip for yucks.

awholegunch commented:
You asked:
`Do we need multiple copies of jquery.js or the cask_404_75pc.png file? Can they not just go under /resources and all the docs that are on the site use them?`
Short answer: We need the multiple copies.
Long answer: Each doc set is independent. They are designed to be zipped and then downloaded locally for local use. So each `current` needs its own copy. They may also change over time, if I ever get around to updating them. 

As for the 404 images: they will probably go, but until we have a working deployment on S3 and I can check how the 404 redirects are working, I would like to leave them, and delete them when that is changed.

Commit:
Consolidate the resources scripts and images into a single resources
directory. Added generation of the redirect-page.js from the different
configuration types. Updated the HTML pages and the favicon.ico files.
Updated README file with warning about the Bamboo error.

wolf31o2 commented on docs.cask.co/scripts/build.sh:
```diff
@@ -18,17 +18,22 @@
 
+__types_array=""
 for __type in ${__types}; do
+  __types_array+=", \"${__type}\""
```
This isn't an array. It's confusing. Also, you're doing this in the loop, where it doesn't make sense. You already have the complete list, with `${types}`, so just use that...

wolf31o2 commented on docs.cask.co/scripts/build.sh:
```diff
@@ -18,17 +18,22 @@
       echo "Could not create 'json-versions.js' and 'version' file for ${__type}" ; exit 1)
 done
+
+sed -e "s|TYPE_ARRAY|${__types_array:1}|" "${__site}/www/resources/redirect-page.js" > "${__site}/target/www/resources/redirect-page.js"  || ( echo "Could not rewrite redirect-page.js to target directory" ; exit 1)
```
```
sed -e "s|TYPE_ARRAY|$(echo ${types} | sed -e 's#^#"#' -e 's# #" "#g' -e 's#$#"#' -e 's# #, #g')|" ...
```

Also, that exit won't exit. It'll exit the subshell that you spawned. Instead, remove the `||` and use:

```
__ret=$?
[[ ${__ret} -ne 0 ]] && echo "..."
exit ${__ret}
```

wolf31o2 replied:
```
$ cat page.js
var something=( TYPE_ARRAY );
sed -e "s|TYPE_ARRAY|$(echo ${types} | sed -e 's#^#"#' -e 's# #" "#g' -e 's#$#"#' -e 's# #, #g')|" page.js
var something=( "cdap", "coopr", "tigon" );
```

awholegunch replied:
Changed. Note that the above needs to be `$(echo ${__types}…`

wolf31o2 commented on docs.cask.co/README.rst:
```diff
@@ -30,6 +30,8 @@ as they are copies used to test the menus when the configuration file has been g
 it is run.
 
+**Note:** Due to https://jira.atlassian.com/browse/BAM-18015, we do not use any sub-directories
+in the ``resources`` directory.
```
Is this necessary? This seems like it would belong in a design doc, or as a comment in a file, not in the main README. I'd remove it.

awholegunch replied:
This README is the design doc for this; we don't have one elsewhere. I wanted to make this prominent, so people won't forget.

wolf31o2 reviewed (changes requested):
After this last set of changes, it 👍 LGTM

Commit:
Fix broken links. Change cdap.css to cask.css, and use the version
in "current" directories for product-specific 404 and index pages.

Commit:
Fix sidebar links.

Commit:
Remove emphasis in sidebar links as it is not being used.

Commit:
Change per comment.

Commit:
Oops. Forgot to grab the return code.

Commit:
Add Google Tag Manager "no script" codes to the top of each <body> section.

Commit:
Add a die function, and use to exit on failures.
Break lines shorter.

wolf31o2 commented:
👍 LGTM
